### PR TITLE
docs: adding an initial CI documentation

### PR DESCRIPTION
**Pull request by beraldoleal**
This is actually a first attempt to document our CI, and all this content was based on the document created by @fidencio (kudos to him). We are just moving the content and discussion from Google Docs to here.

I used the "poetic license" to add some notes on what I believe our CI will look like in the future.

TIP: I suggest viewing the file in preview mode and also enabling the TOC feature in Github for better navigation.

### CODE REVIEW

File: docs/ci.md
Comment (gkurz) [inline docs/ci.md:1]: This link is broken :
```
$ ls -l docs/ci.md 
lrwxrwxrwx. 1 greg greg 12 Feb  1 17:09 docs/ci.md -> ci/README.md
$ realpath docs/ci.md 
realpath: docs/ci.md: No such file or directory
```

Link target should be `../ci/README.md`.

Comment (beraldoleal) [reply docs/ci.md]: fixed. thanks

**Review by fidencio — APPROVED** [review]
lgtm, thanks @beraldoleal!

**Comment by gkurz** [comment]
Hi @beraldoleal ! This still requires the usual `Fixes #####` to pass CI.

**Review by GabyCT — APPROVED** [review]

File: docs/ci.md
Comment (gkurz) [inline docs/ci.md:1]: All markdown files in the `docs` directory must be referenced by some other referenced file or the top level `README.md`. This is mandated by the required static checks :

```
[static-checks.sh:798] ERROR: Document docs/ci.md is not referenced
```

File: ci/README.md
Comment (justxuewei) [inline ci/README.md:167]: I wrote a script to find failed tests in one click. Maybe it is worth displaying here for a quick check. See: https://github.com/kata-containers/kata-containers/discussions/9030.

File: ci/README.md
Comment (gkurz) [inline ci/README.md:307]: Script is https://github.com/kata-containers/kata-containers/blob/main/tools/packaging/kata-deploy/local-build/kata-deploy-build-and-upload-payload.sh .

Comment (beraldoleal) [reply ci/README.md]: fixed.

**Comment by beraldoleal** [comment]
Sorry for the late reply here. Fixed with force-push.

File: docs/ci.md
Comment (beraldoleal) [reply docs/ci.md:1]: Thanks. fixed with the force-push.

**Comment by gkurz** [comment]
@beraldoleal CI fails with :

```
ERROR: Document docs/ci.md is not referenced
````

which is indeed the case AFAICT.

Also, do we really need a symbolic link ?

Cannot you just add a `[Kata Containers CI](../ci/README.md)` line somewhere in `docs/README.md` ?

Or maybe even `[ci](ci/README.md)` at https://github.com/kata-containers/kata-containers/blob/main/README.md?plain=1#L142 ?

**Comment by gkurz** [comment]
\test

File: ci/README.md
Comment (ldoktor) [inline ci/README.md:58]: I know it'd be harder and more prone to changes but it'd be nice to put links to the workflow files at least for the main pipelines so people can quickly find them and see the details.

File: ci/README.md
Comment (ldoktor) [inline ci/README.md:67]: Missing closing bracket

Comment (ldoktor) [reply ci/README.md:67]: Oh it's not missing but the sentence got broken into two items

File: ci/README.md
Comment (ldoktor) [inline ci/README.md:157]: how about adding "please, create an issue for us, and then re-run `(or ask maintainers to re-run)` the tests following these steps" to avoid people without commit right the hassle trying to find out why they can not re-run them?

File: ci/README.md
Comment (ldoktor) [inline ci/README.md:167]: Do you want to mention the re-run all button? From my experience that's what people use the most anyway... (which is sad as)

File: ci/README.md
Comment (ldoktor) [inline ci/README.md:235]: I'd also add `for 15 days`

**Comment by gkurz** [comment]
@beraldoleal I've made this a draft to be sure it doesn't get merged before you had time to look at @ldoktor 's comments.
This doesn't invalidate CI. If you want to proceed without additional changes, you just need to press the 'Ready for review' button.

File: ci/README.md
Comment (beraldoleal) [reply ci/README.md:67]: Nice catch. thanks.

File: ci/README.md
Comment (beraldoleal) [reply ci/README.md:340]: Given, this is a current limitation, I would suggest creating an issue to discuss the workaround there.

File: ci/README.md
Comment (beraldoleal) [reply ci/README.md:58]: Definitely helpful, but I will leave this to others in following PRs. This is just an attempt to move the content from google docs here, because of spam/bots there.

File: ci/README.md
Comment (beraldoleal) [reply ci/README.md:167]: It is mentioned already on line 156. I would keep like that and improve the workflow later.

**Comment by gkurz** [comment]
Unrelated flaky CI blocking the way again. I'll force merge this PR.

**Comment by ldoktor** [comment]
> Unrelated flaky CI blocking the way again. I'll force merge this PR.

Nice, `sudo merge` :-)

**Pull request merged**
